### ISSUE TITLE
Jump to code not working for absolute imports into same package

### ISSUE THREAD

**Issue by schlichtanders** [julia-vscode/LanguageServer.jl#1225]
Hi,

understanding why jump to code does not work is really a hard problem. Now I came up with a good guess I want to share.

When using absolute imports to my own package, the code works (i.e. is valid julia) but jumping to code does not work and the linter reports that it cannot find the functions.

`src/MyPackage.jl`
```julia
module MyPackage
  module SubMod1
    a = 42
  end
  include("file2.jl")
end
```

`src/file2.jl`
```julia
module SubMod2
  using MyPackage.SubMod2: a
end
```

When I run `re-index language server cache` in VSCode, **for a few seconds the code-jumping work**, but then again everything is marked as unresolveable.

**Comment by schlichtanders** [comment]
@pfitzseb can you look over the issue description? 

Maybe you recognize it as a duplicate of another issue, or have a clue where the reason may be for this missing support for submodules.

**Comment by pfitzseb** [comment]
Isn't this just a typo and it should be 
```
module SubMod2
  using MyPackage.SubMod1: a
end
```
in `file2.jl`? That consistently works for me.

**Comment by schlichtanders** [comment]
thank you for the pointer - so this is unfortunately no minimal reproducible example... but just a little typo ... a pity

I tried code jumping again and I am very surprised - it works now with almost everything which hasn't worked for a while.

Only macros don't work - I opened a new ticket for it https://github.com/julia-vscode/LanguageServer.jl/issues/1244

--------

I now made sure to always activate the parent environment, maybe this helped. Before I used julia 1.9 central project to have multiple projects at once. It might be because the environment was just activated that things worked, and things would fail when working longer, but maybe this was really everything which I've done wrong all the time.

It would be great to have autoswitching of environments (I like to open the complete `.julia/dev` folder within vscode).
I opened a ticket at https://github.com/julia-vscode/julia-vscode/issues/3411 respectively
